packages/sdk-metrics/src/aggregator/ExponentialHistogram.ts

```diff
@@ -0,0 +1,606 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Accumulation,
+  AccumulationRecord,
+  Aggregator,
+  AggregatorKind,
+} from './types';
+import {
+  DataPointType,
+  ExponentialHistogramMetricData,
+} from '../export/MetricData';
+import { diag, HrTime } from '@opentelemetry/api';
+import { InstrumentDescriptor, InstrumentType } from '../InstrumentDescriptor';
+import { Maybe } from '../utils';
+import { AggregationTemporality } from '../export/AggregationTemporality';
+import { Buckets } from './exponential-histogram/Buckets';
+import { getMapping } from './exponential-histogram/mapping/getMapping';
+import { Mapping } from './exponential-histogram/mapping/types';
+import * as util from './exponential-histogram//util';
+
+/**
+ * Internal value type for ExponentialHistogramAggregation.
+ * Differs from the exported type as undefined sum/min/max complicate arithmetic
+ * performed by this aggregation, but are required to be undefined in the exported types.
+ */
+interface InternalHistogram {
+  positive: {
+    offset: number;
+    bucketCounts: number[];
+  };
+  negative: {
+    offset: number;
+    bucketCounts: number[];
+  };
+  hasMinMax: boolean;
+  min: number;
+  max: number;
+  sum: number;
+  count: number;
+  scale: number;
+  zeroCount: number;
+}
+
+// HighLow is a utility class used for computing a common scale for
+// two exponential histogram accumulations
+class HighLow {
+  static combine(h1: HighLow, h2: HighLow): HighLow {
+    return new HighLow(Math.min(h1.low, h2.low), Math.max(h1.high, h2.high));
+  }
+  constructor(public low: number, public high: number) {}
+}
+
+const MAX_SCALE = 20;
+const DEFAULT_MAX_SIZE = 160;
+const MIN_MAX_SIZE = 2;
+
+export class ExponentialHistogramAccumulation implements Accumulation {
+  constructor(
+    public startTime: HrTime = startTime,
+    private _maxSize = DEFAULT_MAX_SIZE,
+    private _recordMinMax = true,
+    private _sum = 0,
+    private _count = 0,
+    private _zeroCount = 0,
+    private _min = Number.POSITIVE_INFINITY,
+    private _max = Number.NEGATIVE_INFINITY,
+    private _positive = new Buckets(),
+    private _negative = new Buckets(),
+    private _mapping: Mapping = getMapping(MAX_SCALE)
+  ) {
+    if (this._maxSize < MIN_MAX_SIZE) {
+      diag.warn(`Exponential Histogram Max Size set to ${this._maxSize}, \
+                changing to the minimum size of: ${MIN_MAX_SIZE}`);
+      this._maxSize = MIN_MAX_SIZE;
+    }
+  }
+
+  /**
+   * record updates a histogram with a single count
+   * @param {Number} value
+   */
+  record(value: number) {
+    this.updateByIncrement(value, 1);
+  }
+
+  /**
+   * Sets the start time for this accumulation
+   * @param {HrTime} startTime
+   */
+  setStartTime(startTime: HrTime): void {
```

Comment:
Is there a reason have a setter function instead of just making startTime public or is this just copied from Java?

Reply:
This is part of the [Accumulation](https://github.com/open-telemetry/opentelemetry-js/blob/d1aa906889405f31f7a4a16495aca846f4bed26a/packages/sdk-metrics/src/aggregator/types.ts#L68-L71) interface:

```typescript
export interface Accumulation {
  setStartTime(startTime: HrTime): void;
  record(value: number): void;
}
```